Can't run single tests for rules

if I want to test for a single rule with:

    pytest -k test_rule[missing-whitespace-after-setting]

I get:

    ========================================================================== test session starts ==========================================================================
    platform win32 -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
    rootdir: C:\develop\robot\robotframework-robocop
    collected 220 items / 219 deselected / 1 selected
    
    tests\atest\test_rule.py E                                                                                                                                         [100%]
    
    ================================================================================ ERRORS ================================================================================= 
    _____________________________________________________ ERROR at setup of test_rule[missing-whitespace-after-setting] _____________________________________________________ 
    
        @pytest.fixture
        def robocop_instance():
    >       return Robocop(from_cli=True)
    
    tests\atest\conftest.py:8:
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    robocop\run.py:48: in __init__
        self.config.parse_opts(from_cli=from_cli)
    robocop\config.py:260: in parse_opts
        args = self.parser.parse_args(args)
    C:\Program Files\Python39\lib\argparse.py:1821: in parse_args
        self.error(msg % ' '.join(argv))
    robocop\config.py:95: in error
        super().error(message)
    C:\Program Files\Python39\lib\argparse.py:2575: in error
        self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
    _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    
    self = CustomArgParser(prog='robocop', usage=None, description='Static code analysis tool for Robot Framework', formatter_class=<class 'argparse.RawTextHelpFormatter'>, conflict_handler='error', add_help=False)
    status = 2, message = 'robocop: error: unrecognized arguments: -k\n'
    
        def exit(self, status=0, message=None):
            if message:
                self._print_message(message, _sys.stderr)
    >       _sys.exit(status)
    E       SystemExit: 2
    
    C:\Program Files\Python39\lib\argparse.py:2562: SystemExit
    ------------------------------------------------------------------------- Captured stderr setup ------------------------------------------------------------------------- 
    usage: robocop [-i RULES] [-e RULES] [-rules EXT_RULES] [--no-recursive]
                   [-r REPORTS] [-f FORMAT] [-c CONFIGURABLE] [-l [PATTERN]]
                   [--list-configurables [PATTERN]] [--list-reports] [-o PATH]
                   [--filetypes FILETYPES] [-t THRESHOLD] [-A PATH]
                   [--ignore PATH] [-h] [-v]
                   [paths ...]
    robocop: error: unrecognized arguments: -k
    ======================================================================== short test summary info ======================================================================== 
    ERROR tests/atest/test_rule.py::test_rule[missing-whitespace-after-setting] - SystemExit: 2
    =================================================================== 219 deselected, 1 error in 0.57s ====================================================================

--- a/CONTRIBUTING.rst
+++ b/CONTRIBUTING.rst
@@ -189,6 +189,10 @@ second is path separator - \ under Windows and / under Linux.
 If you wish to use additional configuration or use different that default test data directory follow
 instructions from ``pytest_generate_tests`` method.
 
+In case you want to run only single test (not all dynamically generated ones) you can run::
+
+    pytest --rule rule-name tests\atest
+
 E2E tests
 '''''''''
 

--- a/tests/atest/conftest.py
+++ b/tests/atest/conftest.py
@@ -36,8 +36,7 @@ def pytest_generate_tests(metafunc):
             if rule == selected_rule:
                 break
         else:
-            print(f"Rule: '{selected_rule}' was not found")
-            return
+            pytest.exit(f"Rule: '{selected_rule}' was not found", 1)
         metafunc.parametrize('rule, args, test_data', [(selected_rule, args, test_data)])
         return
     # TODO: load other tests from file (like yaml)

--- a/tests/atest/conftest.py
+++ b/tests/atest/conftest.py
@@ -7,7 +7,11 @@ from robocop.run import Robocop
 
 @pytest.fixture
 def robocop_instance():
-    return Robocop(from_cli=True)
+    return Robocop(from_cli=False)
+
+
+def pytest_addoption(parser):
+    parser.addoption("--rule", action="store")
 
 
 def pytest_generate_tests(metafunc):
@@ -25,6 +29,17 @@ def pytest_generate_tests(metafunc):
     if "rule" not in metafunc.fixturenames:
         return
     auto_discovered_rules = [(rule, None, f"{category}/{rule}") for category, rule in get_rules_for_atest()]
+    selected_rule = metafunc.config.getoption('--rule', None)
+    if selected_rule is not None:
+        # Find and use only selected rule
+        for rule, args, test_data in auto_discovered_rules:
+            if rule == selected_rule:
+                break
+        else:
+            print(f"Rule: '{selected_rule}' was not found")
+            return
+        metafunc.parametrize('rule, args, test_data', [(selected_rule, args, test_data)])
+        return
     # TODO: load other tests from file (like yaml)
     auto_discovered_rules.append((
         'inconsistent-assignment',

--- a/tests/atest/test_rule.py
+++ b/tests/atest/test_rule.py
@@ -6,6 +6,7 @@ from robocop.utils import IS_RF4, DISABLED_IN_4, ENABLED_IN_4
 
 
 def configure_robocop_with_rule(args, runner, rule, path):
+    runner.from_cli = True
     config = Config()
     config.parse_opts([
         '--include',

